SpatialGDK/Source/SpatialGDK/Private/Interop/RPCs/RPCService.cpp

```diff
@@ -0,0 +1,339 @@
+				}
+			}
+		}
+		default:
+			checkNoEntry();
+			break;
+		}
+	}
+}
+
+void RPCService::AdvanceView()
+{
+	AdvanceReceivers();
+	AdvanceSenderQueues();
+}
+
+RPCCallbacks::DataWritten RPCService::MakeDataWriteCallback(TArray<FWorkerComponentData>& OutArray) const
+{
+	return [&OutArray](Worker_EntityId EntityId, Worker_ComponentId ComponentId, Schema_ComponentData* InData) {
+		if (ensure(InData != nullptr))
+		{
+			FWorkerComponentData Data;
+			Data.component_id = ComponentId;
+			Data.schema_type = InData;
+			OutArray.Add(Data);
+		}
+	};
+}
+
+RPCCallbacks::UpdateWritten RPCService::MakeUpdateWriteCallback(TArray<UpdateToSend>& OutUpdates) const
+{
+	return [&OutUpdates](Worker_EntityId EntityId, Worker_ComponentId ComponentId, Schema_ComponentUpdate* InUpdate) {
+		if (ensure(InUpdate != nullptr))
+		{
+			UpdateToSend Update;
+			Update.EntityId = EntityId;
+			Update.Update.component_id = ComponentId;
+			Update.Update.schema_type = InUpdate;
+			OutUpdates.Add(Update);
+		}
+	};
+}
+
+TArray<RPCService::UpdateToSend> RPCService::GetRPCsAndAcksToSend()
+{
+	TArray<UpdateToSend> Updates;
+
+	RPCWritingContext Ctx(RPCWritingContext::DataKind::ComponentUpdate);
+	Ctx.UpdateWrittenCallback = MakeUpdateWriteCallback(Updates);
+
+	Ctx.RPCWrittenCallback = [](Worker_EntityId Entity, Worker_ComponentId ComponentId, uint32 RPCId) {
+		// if (EventTracer != nullptr)
+		//{
+		//	EventTraceUniqueId LinearTraceId = EventTraceUniqueId::GenerateForRPC(EntityId, static_cast<uint8>(Type), NewRPCId);
+		//	FSpatialGDKSpanId SpanId = EventTracer->TraceEvent(FSpatialTraceEventBuilder::CreateSendRPC(LinearTraceId),
+		//		/* Causes */ Payload.SpanId.GetConstId(), /* NumCauses */ 1);
+		//	RPCStore.AddSpanIdForComponentUpdate(EntityComponent, SpanId);
+		//}
+	};
+
+	for (auto& QueueEntry : Queues)
+	{
+		RPCQueueDescription& Queue = QueueEntry.Value;
+		Queue.Queue->FlushAll(Ctx);
+	}
+
+	for (auto& Entry : Receivers)
+	{
+		RPCReceiverDescription& Receiver = Entry.Value;
+		Receiver.Receiver->FlushUpdates(Ctx);
+	}
+
+	return Updates;
+}
+
+TArray<RPCService::UpdateToSend> RPCService::FlushSenderForEntity(Worker_EntityId EntityId)
+{
+	TArray<UpdateToSend> Updates;
+
+	RPCWritingContext Ctx(RPCWritingContext::DataKind::ComponentUpdate);
+	Ctx.UpdateWrittenCallback = MakeUpdateWriteCallback(Updates);
+
+	Ctx.RPCWrittenCallback = [](Worker_EntityId Entity, Worker_ComponentId ComponentId, uint32 RPCId) {
+		// if (EventTracer != nullptr)
+		//{
+		//	EventTraceUniqueId LinearTraceId = EventTraceUniqueId::GenerateForRPC(EntityId, static_cast<uint8>(Type), NewRPCId);
+		//	FSpatialGDKSpanId SpanId = EventTracer->TraceEvent(FSpatialTraceEventBuilder::CreateSendRPC(LinearTraceId),
+		//		/* Causes */ Payload.SpanId.GetConstId(), /* NumCauses */ 1);
+		//	RPCStore.AddSpanIdForComponentUpdate(EntityComponent, SpanId);
+		//}
+	};
+
+	for (auto& QueueEntry : Queues)
+	{
+		RPCQueueDescription& Queue = QueueEntry.Value;
+		Queue.Queue->Flush(EntityId, Ctx);
```

Comment:
Can I see an example implementation of Flush?

Reply:
https://github.com/spatialos/UnrealGDK/pull/2913/files#diff-27f6feab17b1e7732d0ce1e04fb198f54fccbf95f1bcca6eda912312bdce3df5